libtenzir/src/tql2/ast.cpp

```diff
@@ -169,6 +170,51 @@ auto to_operand(const ast::expression& x) -> std::optional<operand> {
     });
 }
 
+// @brief Optimize x > now() +- $y <=> x > $now +- $y and x > now() +- $y
+auto optimize_now(const ast::expression& left,
+                  const relational_operator& rel_op,
+                  const ast::expression& right) -> std::optional<expression> {
+  constexpr auto f = [](const ast::expression& l, const ast::binary_op& op,
+                        const ast::expression& r) -> std::optional<operand> {
+    if (op != ast::binary_op::add and op != ast::binary_op::sub) {
+      return std::nullopt;
+    }
+    // TODO: Evaluate unary_expr to a constant duration
+    auto expr = std::get_if<ast::constant>(r.kind.get());
+    auto call = std::get_if<ast::function_call>(l.kind.get());
+    if (not(expr and call and call->fn.path[0].name == "now")) {
+      return std::nullopt;
+    }
```

Comment:
What is this TODO about?

Reply:
The `TODO` is for cases such as `x > -100d + now()` where the `-100d` would not be a constant but rather a `unary_expr`. This is not evaluated here, as it's a rather general thing.